service/handler.go

```diff
@@ -127,7 +128,7 @@ func (svc *Service) handleMessage(message kafka.Message) (string, error) {
 
 	// PUT request to dataset API to pass the header row and the
 	// number of observations that exist against this job instance
-	if err := instance.PutData(http.DefaultClient); err != nil {
+	if err := instance.PutData(svc.HTTPClient.HTTPClient); err != nil {
```

Comment:
this needs to be `PutData(ctx, svc.HTTPClient)` - the rchttp client alongside the context